git/hooks/pre-commit

```diff
@@ -8,25 +8,17 @@
 # To enable this hook, rename this file to "pre-commit".
 #set -x
 
-List=$(git status | findstr modified | findstr app.config | cut -d ":" -f2 | xargs)
 exitstatus=0
-
-for f in "$List"
+git ls-files --modified --exclude-standard ':/**/app.config' |
```

Comment:
I wasn't aware of the `git ls-files` command before, and it seems very handy for this case.

However, it seems like `git ls-files --modified` won't catch files that are newly added to the repository. This could create a niche issue where someone creates a new project in the solution, has the `app.config` of the project contain an undesired block, and then can commit it to the repo without the pre-hook catching it.

Reply:
I hadn’t thought of that. Good catch!

The simplest solution I can think of is to `git grep` all cached files for forbidden patterns.